prov/sockets/src/sock_ep.c

```diff
@@ -1216,7 +1222,77 @@ int sock_srx_ctx(struct fid_domain *domain,
 	return 0;
 }
 
-static void sock_set_fabric_attr(const struct fi_fabric_attr *hint_attr,
+int sock_get_prefix_len(uint32_t net_addr)
+{
+	int count = 0;
+	while (net_addr > 0) {
+		net_addr = net_addr >> 1;
+		count++;
+	}
+	return count;
+}
+
+char *sock_get_fabric_name(struct sockaddr_in *src_addr)
+{
+	int ret;
+        struct ifaddrs *ifaddrs, *ifa;
+	char *fabric_name = NULL;
+	struct in_addr net_in_addr;
+	struct sockaddr_in *host_addr, *net_addr;
+	char netbuf[35];
+	int prefix_len;
+#if HAVE_GETIFADDRS
+	ret = getifaddrs(&ifaddrs);
+	if (!ret) {
```

Comment:
This would be cleaner to check for an error and abort, rather than indenting the entire rest of the function inside the if clause.